src/plugins/charts/public/plugin.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Plugin, CoreSetup } from 'kibana/public';
+
+import { colorMapsService, ThemeService, ColorsService } from './services';
+
+/** @public */
+export type ColorMapsSetup = typeof colorMapsService;
+
+/** @public */
+export type ColorMapsStart = ColorMapsSetup;
+
+export type Theme = Omit<ThemeService, 'init'>;
+export type Color = Omit<ColorsService, 'init'>;
+
+/** @public */
+export interface ChartsPluginSetup {
+  colorMaps: ColorMapsSetup;
+  colors: Color;
+  theme: Theme;
+}
+
+/** @public */
+export type ChartsPluginStart = ChartsPluginSetup;
+
+/** @public */
+export class ChartsPlugin implements Plugin<ChartsPluginSetup, ChartsPluginStart> {
+  private readonly themeService = new ThemeService();
+  private readonly colorsService = new ColorsService();
+
+  public setup({ uiSettings }: CoreSetup): ChartsPluginSetup {
+    this.themeService.init(uiSettings);
+    this.colorsService.init(uiSettings);
+
+    return {
+      colorMaps: colorMapsService,
```

Comment:
You are exposing the color maps service via the contract, but it looks like it is just offering static helper functions and is not tied to any internal state of the plugin - is this correct? If yes you can just re-export it from the top level `index.ts` of the `charts` plugin and import where needed. Then [this problem](https://github.com/elastic/kibana/pull/55469/files#diff-0c882d9cf3550e3ebe630bac56740141R45) is also solved  directly.